Public preview documentation for sponsors (#21166)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/user-list-sponsors.md

---
title: "List sponsors"

> description: "Get user's sponsors. Returns the users or groups assigned as the user's sponsors."

Reviewer: ```suggestion
description: "Get a guest user's sponsors."
```

ms.localizationpriority: medium
author: "sadia353"
ms.prod: "users"
doc_type: apiPageType
---

# List sponsors

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Get user's sponsors. Returns the users or groups assigned as the user's sponsors.

Reviewer: ```suggestion
Get a guest user's sponsors. Sponsors are users and groups that are responsible for this guest user's privileges in the tenant and for keeping the guest user's information and access up to date.
```

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged)                                    |
| :------------------------------------- | :----------------------------------------------------------------------------- |

> | Delegated (work or school account)     | User.Read.All, User.ReadWrite.All, Directory.Read.All, Directory.ReadWrite.All |
> | Delegated (personal Microsoft account) | Not supported.                                                                 |
> | Application                            | User.Read.All, User.ReadWrite.All, Directory.Read.All, Directory.ReadWrite.All |

Reviewer: Let's avoid documenting Directory.* permissions because they are too privileged, unless they are the only ones supported.
```suggestion
| Delegated (work or school account)     | User.Read.All, User.ReadWrite.All |
| Delegated (personal Microsoft account) | Not supported.                                                                 |
| Application                            | User.Read.All, User.ReadWrite.All|
```

[!INCLUDE [limited-info](../../includes/limited-info.md)]

## HTTP request

> Get sponsors:

Reviewer: ```suggestion
Get sponsors.
```

<!-- { "blockType": "ignored" } -->
```http
GET /users/{id | userPrincipalName}/sponsors
```

## Optional query parameters

> This method supports the `$select` and `$expand` [OData query parameters](/graph/query-parameters) to help customize the response.  
> 
> >**Note:**
> 
> > + You can specify `$select` inside `$expand` to select the individual sponsor's properties: `$expand=sponsors($select=id,displayName)`.

Reviewer: Great call out of nested $select!
No need for a note.  A regular sentence will do.
```suggestion
This method supports the `$select` and `$expand` [OData query parameters](/graph/query-parameters) to help customize the response. You can specify `$select` inside `$expand` to select the individual sponsor's properties: `$expand=sponsors($select=id,displayName)`.
```

## Request headers

| Header           | Value                                                                                          |
| :--------------- | :--------------------------------------------------------------------------------------------- |
| Authorization    | Bearer {token}. Required.                                                                      |

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `200 OK` response code and a collection of [user](../resources/user.md) and [group](../resources/group.md) objects in the response body.

## Examples

### Example: Get sponsors

The following example shows a request to get the sponsors.

#### Request

<!-- {
  "blockType": "request",
  "name": "get_sponsor"
}-->

``` http

> GET https://graph.microsoft.com/beta/users/{user-id}/sponsors

Reviewer: nit: You can replace `{user-id}` with a real-looking user ID.
